File: flutter_app/lib/models/deal/deal.dart
class Deal {
  final int dealOrder;
  final String imageUrl;
  final String title;
  final String? description;
  final String action;
  final String aspectRatio;

  Deal({
    required this.dealOrder,
    required this.imageUrl,
    required this.title,
    this.description,
    required this.action,
    required this.aspectRatio,
  });

  factory Deal.fromJson(Map<String, dynamic> json) {
    return Deal(
      dealOrder: json['dealOrder'],
      imageUrl: json['imageUrl'],
      title: json['title'],
      description: json['description'],
      action: json['action'],
      aspectRatio: json['aspectRatio'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'dealOrder': dealOrder,
      'imageUrl': imageUrl,
      'title': title,
      'description': description,
      'action': action,
      'aspectRatio': aspectRatio,
    };
  }

  double get aspectRatioValue {
    final parts = aspectRatio.split(':');
    if (parts.length == 2) {
      final width = double.tryParse(parts[0]) ?? 1;
      final height = double.tryParse(parts[1]) ?? 1;
      return width / height;
    }
    return 1.0;
  }
}

List<Deal> parseDeals(dynamic jsonList) {
  return (jsonList as List).map((json) => Deal.fromJson(json)).toList();
}

File: backend/models/Deal.js
const mongoose = require('mongoose')

const DealSchema = new mongoose.Schema(
  {
    dealOrder: {
      type: Number,
      required: true,
      unique: true
    },
    imageUrl: {
      type: String,
      required: true,
    },
    title: {
      type: String,
      required: true
    },
    description: {
      type: String
    },
    action: {
      type: String,
      required: true,
    },
    aspectRatio: {
      type: String,
      required: true,
      enum: ['16:9', '4:3', '3:2'],
      default: '16:9',
    },
  },
  { timestamps: true }
)

module.exports = mongoose.model('Deal', DealSchema)